src/NuGet.Core/NuGet.Commands/RestoreCommand/Utility/AuditUtility.cs

```diff
@@ -78,10 +82,27 @@ public AuditUtility(
             }
         }
 
+        private void CheckPackageDownloadVulnerabilities(IReadOnlyList<IReadOnlyDictionary<string, IReadOnlyList<PackageVulnerabilityInfo>>> knownVulnerabilities)
+        {
+            foreach (var targetFramework in _targetFrameworks)
+            {
+                foreach (var downloadDependency in targetFramework.DownloadDependencies)
+                {
+                    List<PackageVulnerabilityInfo>? knownVulnerabilitiesForPackage = GetKnownVulnerabilities(downloadDependency.Name, downloadDependency.VersionRange.MaxVersion!, knownVulnerabilities);
+
+                    if (knownVulnerabilitiesForPackage?.Count > 0)
+                    {
+                        PackageDownloadPackagesWithAdvisory ??= new();
+                        PackageDownloadPackagesWithAdvisory.Add(downloadDependency.Name);
+                    }
+                }
+            }
+        }
+
         public async Task<bool> CheckPackageVulnerabilitiesAsync(CancellationToken cancellationToken)
         {
             // Performance: Early exit if restore graph does not contain any packages.
-            if (!HasPackages())
+            if (!HasPackages() && (_targetFrameworks.Count() == 0))
```

Comment:
never use use Linq's `.Count()` when the type has a `.Length` or `.Count` property.